calculator/lexer.py

```diff
@@ -11,8 +72,85 @@ def lex(s):
       * "*" - Multiplication
       * "+" - Addition
       * "-" - Subtraction
+      * "/" - Division
     * The following grouping symbols are supported:
       * "("
       * ")"
     """
-    return []
+
+    # Defining char_before and char_after
+    char_before = None
+    char_after = None
+    # Partial_num is the var that stores my nums until I have to append it to my tokens list
+    partial_num = None
+    # List that stores the tokens
+    tokens = []
+    # Stripping the input line of spaces
+    stripped_line = line
+
+    # Iterating through each index in my stripped_line
+    for i in range(0, len(stripped_line)):
+        # Defining "c" as the char at the index "i" in stripped_line
+        c = stripped_line[i]
+        # Defining char_before but if i is the first index then the index will be invalid
+        if i != 0:
+            # Char_before is the current char's index (i), but subtracting 1 because it is 1 before
+            char_before = stripped_line[i - 1]
+        else:
+            # if i is the first index then char_before is None
+            char_before = None
+
+        # Catching the same error as the first if statement but if i is the last index
+        if i != len(stripped_line) - 1:
+            # Char_after is the current char's index (i), but adding 1 because it is 1 after
+            char_after = stripped_line[i + 1]
+        else:
+            # If i is the last index then char_after is None
+            char_after = None
+        # If c (current char), is a digit or a decimal point. If c is a int or a float:
+        if is_numeric_char(c):
+            # If c is the first iteration:
+            if partial_num == None:
+                # Partial_num is equal to c instead of partial_num is the original value but c added on.
+                partial_num = str(c)
+            else:
+                partial_num += str(c)
+            # If the char_after is a operator or a parenthesis:
```

Comment:
This comment is no longer correct given the most recent refactorings.

Reply:
In this case, I like the following style:
```
# The following conditional is going to determine if we've seen all of a number (float or otherwise) 
if (
                is_op(char_after) # If we see an op afterwards, we know we're done with the number
                or is_parenthesis(char_after) # or a parenthesis
                or char_after == " " # or if the character after is a space (the natural breaking point of a number in math)
                or i == len(stripped_line) - 1 # or if we've hit the end of line
            ):
```

Reply:
Also, just like I commented before about `is_first_iteration`, you might want an `is_end_of_line` variable, which would make `i == len(stripped_line) - 1` into `is_end_of_line`, which is clearer.

Reply:
👍